packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/DatasetController.java

```diff
@@ -220,11 +222,24 @@ public ResponseEntity<Dataset> createDataset(
 	public ResponseEntity<Dataset> getDataset(
 			@PathVariable("id") final UUID id,
 			@RequestParam(name = "project-id", required = false) final UUID projectId) {
-		final Schema.Permission permission =
-				projectService.checkPermissionCanWrite(currentUserService.get().getId(), projectId);
+		final Schema.Permission permission = projectId == null
+				? Schema.Permission.NONE
+				: projectService.checkPermissionCanWrite(
+						currentUserService.get().getId(), projectId);
 
 		try {
 			final Optional<Dataset> dataset = datasetService.getAsset(id, permission);
+
+			if (dataset.isEmpty()) {
+				return ResponseEntity.noContent().build();
+			}
+			// GETs not associated to a projectId cannot read non public or temporary assets
+			if (permission.equals(Schema.Permission.NONE)
+					&& !dataset.get().getPublicAsset()
+					&& !dataset.get().getTemporary()) {
+				throw new ResponseStatusException(HttpStatus.FORBIDDEN, messages.get("rebac.unauthorized-update"));
```

Comment:
I think this error message is meant to be `rebac.unauthorized-read`

Reply:
oh - yes, though it would appear that a 500 is returned and not a 403 but that is details

Reply:
err sorry, i meant the call to `messages.get` here will return

> You are not authorized to update this resource. Please contact the project owner to upgrade permissions.